src/iter.rs

```diff
@@ -57,11 +57,97 @@ where
     }
 }
 
+#[derive(Debug)]
+struct RingBuf<const SIZE: usize = 10> {
+    vals: [u64; SIZE],
+    head: u8,
+    max: u64,
+    maxpos: u8,
+}
+
+impl<const SIZE: usize> RingBuf<SIZE> {
+    fn new() -> Self {
+        assert!(SIZE <= u8::MAX.into());
+        assert!(SIZE > 0);
+        Self {
+            vals: [0; SIZE],
+            head: 0,
+            maxpos: 0,
+            max: 0,
+        }
+    }
+
+    fn update(&mut self, new: u64) {
+        // % here is a purely for the optimizer
+        let head: usize = usize::from(self.head) % self.vals.len();
+        self.vals[head] = new;
+
+        if new > self.max {
+            // this is now the new maximum
+            self.maxpos = self.head;
+            self.max = new;
+        } else if self.maxpos == self.head && new < self.max {
+            // this was the maximum and may not be anymore
+            let (idx, val) = self
+                .vals
+                .iter()
+                .enumerate()
+                .max()
+                .expect("array has fixded size > 0");
+            self.maxpos = idx.try_into().unwrap();
+            self.max = *val;
+        }
+
+        self.head = (self.head + 1) % (self.vals.len() as u8);
+    }
+
+    fn max(&self) -> u64 {
+        self.max
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct RingBufWrap<const RESET: u8 = 5, const BUFSIZE: usize = 10> {
+    buf: Option<(Box<RingBuf<BUFSIZE>>, u8)>,
+}
+
+impl<const RESET: u8, const BUFSIZE: usize> RingBufWrap<RESET, BUFSIZE> {
+    pub(crate) fn new() -> Self {
+        Self { buf: None }
+    }
+    fn update_seq(&mut self, oldpos: u64, delta: u64) -> u64 {
```

Comment:
Nits: empty line for the module. As above wrt the `const` generics. Use let-else with early return to reduce rightward drift in this method.